### PR TITLE
Implement execution of chat ClickEvents

### PR DESCRIPTION
**Pull request by einrobin**
Adds a method to the ClickEvent for chat components to execute it just like a user would click on it in the chat.

Additionally, there is the ChatInputChangeEvent that will be fired when the chat input is changed through a click event or something else other than the direct user input.

### CODE REVIEW

File: mcapi/src/main/java/net/flintmc/mcapi/chat/ChatController.java [hunk 1]
```diff
@@ -43,6 +44,16 @@
    */
   boolean dispatchChatInput(String message);
 
+  /**
+   * Sets the current value of the text field in the chat and fires the {@link
+   * ChatInputChangeEvent}, if the chat is currently not opened, only this event will be fired.
```
Comment (Janrupf) [inline mcapi/src/main/java/net/flintmc/mcapi/chat/ChatController.java]: This  sentence is hard to understand, try making it more clear by splitting it.

**Review by juliarn — APPROVED** [review]

File: mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/chat/component/event/VersionedOpenUrlActionExecutor.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import net.minecraft.client.gui.screen.ConfirmOpenLinkScreen;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.util.Util;
+import org.apache.logging.log4j.Logger;
+
+@Singleton
+@ClickEventAction(value = Action.OPEN_URL, priority = 0)
+public class VersionedOpenUrlActionExecutor implements ClickActionExecutor {
+
+  private static final Set<String> ALLOWED_PROTOCOLS = Sets.newHashSet("http", "https");
```
Comment (Janrupf) [inline mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/chat/component/event/VersionedOpenUrlActionExecutor.java:43]: Maybe we could allow adding additional protocols which can then be handled by packages?

File: mcapi/src/v1_16_5/java/net/flintmc/mcapi/v1_16_5/chat/component/event/VersionedClipboardActionExecutor.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import com.google.inject.Singleton;
+import net.flintmc.mcapi.chat.component.event.ClickActionExecutor;
+import net.flintmc.mcapi.chat.component.event.ClickEvent;
+import net.flintmc.mcapi.chat.component.event.ClickEvent.Action;
+import net.flintmc.mcapi.chat.component.event.ClickEventAction;
+import net.minecraft.client.Minecraft;
+
+@Singleton
+@ClickEventAction(value = Action.COPY_TO_CLIPBOARD, priority = -1)
```
Comment (Janrupf) [inline mcapi/src/v1_16_5/java/net/flintmc/mcapi/v1_16_5/chat/component/event/VersionedClipboardActionExecutor.java]: Is there a reason `priority` is `-1`? The default priority seems to be `1`, so `0` should be fine for this this event

File: mcapi/src/v1_16_5/java/net/flintmc/mcapi/v1_16_5/chat/component/event/VersionedOpenUrlActionExecutor.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import net.minecraft.client.gui.screen.ConfirmOpenLinkScreen;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.util.Util;
+import org.apache.logging.log4j.Logger;
+
+@Singleton
+@ClickEventAction(value = Action.OPEN_URL, priority = 0)
+public class VersionedOpenUrlActionExecutor implements ClickActionExecutor {
+
+  private static final Set<String> ALLOWED_PROTOCOLS = Sets.newHashSet("http", "https");
```
Comment (Janrupf) [inline mcapi/src/v1_16_5/java/net/flintmc/mcapi/v1_16_5/chat/component/event/VersionedOpenUrlActionExecutor.java:43]: See comment on 1.15.2 implementation

**Review by Janrupf — CHANGES_REQUESTED** [review]

File: mcapi/src/main/java/net/flintmc/mcapi/chat/component/event/ClickActionExecutor.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package net.flintmc.mcapi.chat.component.event;
+
+/**
+ * Executor for a {@link ClickEvent}.
+ * <p>
+ * This interface should be used together with the {@link ClickEventAction} annotation to
```
Comment (juliarn) [inline mcapi/src/main/java/net/flintmc/mcapi/chat/component/event/ClickActionExecutor.java]: Maybe change this to ``an implementation of this interface``, the current sentence might be confusing.

**Review by juliarn — CHANGES_REQUESTED** [review]

File: mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/chat/component/event/VersionedOpenUrlActionExecutor.java (hunk 2 above)
Comment (Janrupf) [reply mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/chat/component/event/VersionedOpenUrlActionExecutor.java:43]: Decided to be done later

File: mcapi/src/v1_16_5/java/net/flintmc/mcapi/v1_16_5/chat/component/event/VersionedOpenUrlActionExecutor.java (hunk 4 above)
Comment (Janrupf) [reply mcapi/src/v1_16_5/java/net/flintmc/mcapi/v1_16_5/chat/component/event/VersionedOpenUrlActionExecutor.java:43]: Decided to be done later

**Review by Janrupf — APPROVED** [review]

**Review by juliarn — APPROVED** [review]

**Pull request merged**